examples/observation_collection_for_imitation_learning.py

```diff
@@ -30,12 +30,19 @@ def main(scenarios, headless, seed):
 
     smarts.reset(next(scenarios_iterator))
 
+    prev_vehicles = set()
+    done_vehicles = set()
     for _ in range(5000):
         smarts.step({})
-        smarts.attach_sensors_to_vehicles(
-            agent_spec, smarts.vehicle_index.social_vehicle_ids()
-        )
-        obs, _, _, _ = smarts.observe_from(smarts.vehicle_index.social_vehicle_ids())
+
+        current_vehicles = smarts.vehicle_index.social_vehicle_ids()
+        done_vehicles = prev_vehicles - current_vehicles
+        prev_vehicles = current_vehicles
+
+        smarts.attach_sensors_to_vehicles(agent_spec, current_vehicles)
+        obs, _, _, dones = smarts.observe_from(current_vehicles)
+        for v in done_vehicles:
+            dones[f"Agent-{v}"] = True
```

Comment:
Sorry, I am not going to block with this but I am unsure what we are doing with dones here. Why are we combining the dones of the previous step with the dones of the current step observe?

Reply:
The `dones` returned from `observe_from()` do not include vehicles that were removed from the simulation because their trajectory (from the traffic history db) was completed, it only includes vehicles that are done because they met one of the "done criteria" (off road, collision, etc.).  

So if we want to include vehicles that have finished playing out their history, we have to explicitly check for that.  One way -- probably the "proper" way --  would have been to use a different `AgentInterface` for each vehicle, and set its `max_episode_steps` accordingly.   However, I didn't like this because it made the code around the `attach_sensors_to_vehicles()` call ugly, so I opted for the above instead.

What's happening there is that we're explicitly watching for which vehicles left the simulation _since_ the previous step and marking them `done`.  The `dones` list returned by `observe_from()` _should_ be empty on each step, given that all vehicles in the history data stay on the road and there aren't any collisions (afaik).  I originally asserted that, but then we discovered that the map is currently incorrect so we occasionally still have a vehicles show up as off-road (until I fix the map, at which point I'll add the assert).

Reply:
I'm going to add a comment to try to make this clearer (given that this is an "example"!).